global_helpers/panther_box_helpers.py

```diff
@@ -0,0 +1,123 @@
+    """Error returned when a box item lookup fails."""
+
+
+class BadSecretsLookup(Exception):
+    """Error returned when an AWS Secrets Manager lookup fails."""
+
+
+def is_box_sdk_enabled() -> bool:
+    return Client and JWTAuth
+
+
+def lookup_box_file(user_id: str, file_id: str) -> dict:
+    try:
+        client = get_box_client()
+        user = client.user(user_id=user_id)
+        file_info = client.as_user(user).file(file_id=file_id).get()
+        return file_info
+    except BoxAPIException as err:
+        raise BadBoxLookup('Exception looking up file info.') from err
+
+
+def lookup_box_folder(user_id: str, folder_id: str) -> dict:
+    try:
+        client = get_box_client()
+        user = client.user(user_id=user_id)
+        folder = client.as_user(user).folder(folder_id=folder_id).get()
+        return folder
+    except BoxAPIException as err:
+        raise BadBoxLookup('Exception looking up folder info.') from err
+
+
+def get_box_client() -> Client:
+    # pylint: disable=global-statement
+    global BOX_ACCESS_AGE
+    global BOX_CLIENT
+    if not Client or not JWTAuth:
+        raise Exception('Could not import necessary Box Library.')
+    if BOX_CLIENT is not None and BOX_ACCESS_AGE is not None and datetime.now(
+    ) - BOX_ACCESS_AGE < timedelta(minutes=MAX_BOX_ACCESS_AGE):
+        return BOX_CLIENT
+    response = boto3.client('secretsmanager').get_secret_value(
+        SecretId=BOX_API_ACCESS_NAME)
+    settings = build_jwt_settings(response)
+    BOX_CLIENT = Client(JWTAuth.from_settings_dictionary(settings))
+    BOX_ACCESS_AGE = datetime.now()
+    return BOX_CLIENT
+
+
+def build_jwt_settings(response: dict) -> dict:
+    data = None
+    if 'SecretString' in response:
+        data = response.get('SecretString')
+    else:
+        data = base64.b64decode(response.get('SecretBinary', '{}'))
+    # convert str from aws secrets mgr to json
+    data = json.loads(data)
+    # check that all necessary secrets are configured
+    expected_keys = {
+        BOX_CLIENT_ID, BOX_CLIENT_SECRET, BOX_JWT_PRIVATE_KEY,
+        BOX_JWT_PUB_KEY_ID, BOX_ENTERPRISE_ID, BOX_JWT_KEY_PASSPHRASE
+    }
+    missing_keys = expected_keys - set(data.keys())
+    if len(missing_keys) != 0:
+        raise BadSecretsLookup(
+            'Missing necessary secret(s): {}'.format(missing_keys))
+    # build box jwt settings from gathered secrets
+    settings = {
+        "boxAppSettings": {
+            "clientID": data[BOX_CLIENT_ID],
+            "clientSecret": data[BOX_CLIENT_SECRET],
+            "appAuth": {
+                "publicKeyID": data[BOX_JWT_PUB_KEY_ID],
+                # handling of escaped newlines when dealing with
+                #   secrets mgr -> str -> json.loads()
+                "privateKey": data[BOX_JWT_PRIVATE_KEY].replace('\\n', '\n'),
+                "passphrase": data[BOX_JWT_KEY_PASSPHRASE],
+            },
+        },
+        "enterpriseID": data[BOX_ENTERPRISE_ID],
```

Comment:
These should use the .get method to retrieve the dictionary value i.e. `data.get(BOX_CLIENT_ID)` if you're okay with it returning `None` if it is not present. You avoid a KeyError exception this way.

Reply:
I didn't originally use that method here since it has already validated the existence of these keys in the logic right before - but I'll update it to be consistent with how we normally do dictionary access. 👍🏻